lib/services/local/saml_idp_service_provider.go

```diff
@@ -112,13 +112,14 @@ func (s *SAMLIdPServiceProviderService) GetSAMLIdPServiceProvider(ctx context.Co
 // CreateSAMLIdPServiceProvider creates a new SAML IdP service provider resource.
 func (s *SAMLIdPServiceProviderService) CreateSAMLIdPServiceProvider(ctx context.Context, sp types.SAMLIdPServiceProvider) error {
 	if sp.GetEntityDescriptor() == "" {
-		if err := s.fetchAndSetEntityDescriptor(sp); err != nil {
+		// fetchAndSetEntityDescriptor is expected to return error if it fails
+		// to set entity descriptor. Let's still be defensive and double check
+		// for sp.GetEntityDescriptor() value.
+		if err := s.fetchAndSetEntityDescriptor(sp); err != nil || sp.GetEntityDescriptor() == "" {
 			// We aren't interested in checking error type as any occurrence of error mean entity descriptor was not set.
-			// But a debug log should be helpful to indicate source of error.
-			s.log.Debugf("Failed to fetch entity descriptor from %s. %s.", sp.GetEntityID(), err.Error())
```

Comment:
FWIW I think this log entry could've been useful in troubleshooting.

Reply:
brought logger back in https://github.com/gravitational/teleport/pull/38825/commits/ee1b015ce0448c3684b544398e20d1aef21e0aa4